app/gui/suggestion-database/src/entry.rs

```diff
@@ -604,17 +595,17 @@ impl Entry {
             })
         }
 
-        let (documentation, icon_name, doc_sections) = match &mut entry {
-            Type { documentation, documentation_html, documentation_sections, .. }
-            | Constructor { documentation, documentation_html, documentation_sections, .. }
-            | Method { documentation, documentation_html, documentation_sections, .. }
-            | Module { documentation, documentation_html, documentation_sections, .. }
-            | Function { documentation, documentation_html, documentation_sections, .. }
-            | Local { documentation, documentation_html, documentation_sections, .. } => {
-                let documentation =
-                    Self::make_html_docs(mem::take(documentation), mem::take(documentation_html));
-                let icon_name = find_icon_name_in_doc_sections(&*documentation_sections);
-                (documentation, icon_name, mem::take(documentation_sections))
+        let (icon_name, doc_sections) = match &entry {
+            Type { documentation, .. }
+            | Constructor { documentation, .. }
+            | Method { documentation, .. }
+            | Module { documentation, .. }
+            | Function { documentation, .. }
+            | Local { documentation, .. } => {
+                let docs = documentation.as_ref().map(|s| s.as_ref()).unwrap_or_default();
+                let documentation_sections = enso_doc_parser::parse(docs);
+                let icon_name = find_icon_name_in_doc_sections(&documentation_sections);
+                (icon_name, documentation_sections)
```

Comment:
As we now own `documentation_sections`, we can read icon_name outside the match, simplifying the code.